pkg/cleanup/containers.go

```diff
@@ -134,19 +135,20 @@ func (c *containers) stopContainerd() {
 
 func (c *containers) stopAllContainers() error {
 	var errs []error
-	logrus.Debugf("trying to list all pods")
 
 	var pods []string
+	ctx := context.Background()
 	err := retry.Do(func() error {
+		logrus.Debugf("trying to list all pods")
 		var err error
-		pods, err = c.Config.containerRuntime.ListContainers()
+		pods, err = c.Config.containerRuntime.ListContainers(ctx)
 		if err != nil {
 			return err
 		}
 		return nil
-	})
+	}, retry.Context(ctx))
```

Comment:
Another useful option in this context might be LastErrorOnly, otherwise each error is collected and reported individually.

```suggestion
	}, retry.Context(ctx), retry.LastErrorOnly(true))
```